jmxtrans-output/jmxtrans-output-core/src/main/java/com/googlecode/jmxtrans/model/output/StatsDTelegrafWriter.java

```
@@ -0,0 +1,134 @@
+			resultTagList.add(",jmxport=" + server.getPort());
+			//tagList.add("objectName=" + query.getObjectName());
+			resultTagList.add("attribute=" + attributeName);
+
+			for (Map.Entry<String, Object> values : result.getValues().entrySet()) {
+
+				String field = values.getKey();
+				Object value = values.getValue();
+				if (isNotValidValue(value)) {
+					log.debug("Skipping message key[{}] with value: {}.", field, value);
+					continue;
+				}
+				List<String> tagList = new ArrayList(resultTagList);
+
+				boolean isSingleValueAttribute = StringUtils.equals(attributeName, field);
+				if( !isSingleValueAttribute ){
+					tagList.add("resultKey="+values.getKey());
+				}
+
+				for (Map.Entry e : tags.entrySet()) {
+					tagList.add(e.getKey() + "=" + e.getValue());
+				}
+
+				Number actualValue = computeActualValue(value);
+				StringBuilder sb = new StringBuilder(result.getKeyAlias())
+					.append(StringUtils.join(tagList, ","))
+					.append(":").append(actualValue)
+					.append("|").append(bucketType).append("\n");
+
+				String output = sb.toString();
+				if( actualValue.floatValue() < 0 && !StatsDMetricType.GAUGE.getKey().equals(bucketType) )
+				{
+					log.debug("Negative values are only supported for gauges, not sending: {}.", output);
+				}
+				else{
+					log.debug(output);
+					writer.write(output);
+				}
+			}
+		}
+	}
+
+	private String getBucketType(int resultIndex) {
+		if( this.bucketTypes.length > resultIndex ){
+			return bucketTypes[resultIndex];
+		}
+		return bucketTypes[bucketTypes.length - 1];
+	}
+
+	private boolean isNotValidValue(Object value) {
+		return !(isNumeric(value) /*|| stringsValuesAsKey*/);
+	}
+
+	private Number computeActualValue(Object value) {
+		Object transformedValue = valueTransformer.apply(value);
+		if (isNumeric(transformedValue)) {
+			return transformedValue instanceof Number ?
+				(Number) transformedValue : Float.valueOf(transformedValue.toString());
+		}
+		return null;
```

Since this class only works with Numbers, we should throw an exception instead of returning `null`.